Tests/SentryTests/Helper/SentryFileManagerTests.swift

```diff
@@ -139,7 +139,49 @@ class SentryFileManagerTests: XCTestCase {
         let actualData = envelopes[0].contents
         XCTAssertEqual(expectedData, actualData as Data)
     }
-    
+
+    func testDeleteOldEnvelopes() throws {
+        let envelope = TestConstants.envelope
+        let path = sut.store(envelope)
+
+        let date = Date(timeIntervalSinceNow: -90 * 24 * 60 * 60)
+        try FileManager.default.setAttributes([FileAttributeKey.creationDate: date], ofItemAtPath: path)
+
+        XCTAssertEqual(sut.getAllEnvelopes().count, 1)
+
+        sut.deleteOldEnvelopesFromAllSentryPaths()
+
+        XCTAssertEqual(sut.getAllEnvelopes().count, 0)
+    }
+
+    func testDontDeleteYoungEnvelopes() throws {
+        let envelope = TestConstants.envelope
+        let path = sut.store(envelope)
+
+        let date = Date(timeIntervalSinceNow: (-90 * 24 * 60 * 60) + 1)
```

Comment:
If you want to be even more precise you can change this to, and then for sure your test would start failing some times when using NSDate in the implementation.
```suggestion
        let date = Date(timeIntervalSinceNow: (-90 * 24 * 60 * 60) + 0.001)
```